win32ss/user/user32/controls/ghost.c

```diff
@@ -0,0 +1,594 @@
+    HDC hdc = BeginPaint(hwnd, &ps);
+    if (hdc)
+    {
+        // don't draw at here
+        EndPaint(hwnd, &ps);
+    }
+}
+
+static void
+Ghost_OnMove(HWND hwnd, int x, int y)
+{
+    RECT rc;
+    HWND hwndTarget = Ghost_GetTarget(hwnd);
+
+    GetWindowRect(hwnd, &rc);
+
+    // move the target
+    SetWindowPos(hwndTarget, NULL, rc.left, rc.top, 0, 0,
+        SWP_NOSIZE | SWP_NOOWNERZORDER | SWP_NOACTIVATE | SWP_NOSENDCHANGING);
+}
+
+static void
+Ghost_OnDestroy(HWND hwnd)
+{
+    KillTimer(hwnd, GHOST_TIMER_ID);
+}
+
+static void
+Ghost_DestroyTarget(GHOST_DATA *pData)
+{
+    HWND hwndTarget = pData->hwndTarget;
+    DWORD pid;
+    HANDLE hProcess;
+
+    pData->hwndTarget = NULL;
+    GetWindowThreadProcessId(hwndTarget, &pid);
+
+    hProcess = OpenProcess(PROCESS_TERMINATE, FALSE, pid);
+    if (hProcess)
+    {
+        TerminateProcess(hProcess, -1);
+        CloseHandle(hProcess);
+    }
+
+    DestroyWindow(hwndTarget);
+}
+
+static void
+Ghost_OnNCDestroy(HWND hwnd)
+{
+    // delete the user data
+    GHOST_DATA *pData = Ghost_GetData(hwnd);
+    if (pData)
+    {
+        SetWindowLongPtrW(hwnd, GWLP_USERDATA, 0);
+
+        // delete image
+        DeleteObject(pData->hbm32bpp);
+        pData->hbm32bpp = NULL;
+
+        // remove prop
+        RemovePropW(pData->hwndTarget, GHOST_PROP);
+
+        // show target
+        ShowWindowAsync(pData->hwndTarget, SW_SHOWNOACTIVATE);
+
+        // destroy target if necessary
+        if (pData->bDestroyTarget)
+        {
+            Ghost_DestroyTarget(pData);
+        }
+
+        HeapFree(GetProcessHeap(), 0, pData);
+    }
+
+    NtUserSetWindowFNID(hwnd, FNID_DESTROY);
+
+    PostQuitMessage(0);
+}
+
+static void
+Ghost_OnClose(HWND hwnd)
+{
+    INT id;
+    WCHAR szAskTerminate[128];
+    WCHAR szHungUpTitle[128];
+
+    // stop timer
+    KillTimer(hwnd, GHOST_TIMER_ID);
+
+    LoadStringW(User32Instance, IDS_ASK_TERMINATE,
+                szAskTerminate, ARRAYSIZE(szAskTerminate));
+    LoadStringW(User32Instance, IDS_HUNG_UP_TITLE,
+                szHungUpTitle, ARRAYSIZE(szHungUpTitle));
+
+    id = MessageBoxW(hwnd, szAskTerminate, szHungUpTitle,
+                     MB_ICONINFORMATION | MB_YESNO);
+    if (id == IDYES)
+    {
+        // destroy the target
+        Ghost_Unenchant(hwnd, TRUE);
+        return;
+    }
+
+    // restart timer
+    SetTimer(hwnd, GHOST_TIMER_ID, GHOST_INTERVAL, NULL);
+}
+
+static void
+Ghost_OnTimer(HWND hwnd, UINT id)
+{
+    HWND hwndTarget;
+    DWORD_PTR dwResult;
+    DWORD dwTimeout;
+    DWORD dwTick1, dwTick2;
+    GHOST_DATA *pData = Ghost_GetData(hwnd);
+
+    if (id != GHOST_TIMER_ID || !pData)
+        return;
+
+    // stop the timer
+    KillTimer(hwnd, id);
+
+    hwndTarget = pData->hwndTarget;
+    if (!IsWindow(hwndTarget) || IsIconic(hwnd) || IsIconic(hwndTarget))
+    {
+        // resume if window is destroyed
+        Ghost_Unenchant(hwnd, FALSE);
+        return;
+    }
+
+    // resume if responding
```

Comment:
Why don't you use [IsHungAppWindow](https://github.com/reactos/reactos/blob/master/win32ss/user/user32/windows/window.c#L1842)?